client/components/tinymce/plugins/simple-payments/dialog/index.jsx

```diff
@@ -43,6 +41,50 @@ import UpgradeNudge from 'my-sites/upgrade-nudge';
 // Utility function for checking the state of the Payment Buttons list
 const isEmptyArray = a => Array.isArray( a ) && a.length === 0;
 
+// Selector to get the form values, insert the missing field, and convert it to
+// a custom post data structure ready to be passed to `wpcom` API.
+const getPostFromForm = state => {
+	const currencyCode = getCurrentUserCurrencyCode( state );
+	const formValues = getProductFormValues( state );
+
+	if ( currencyCode ) {
+		formValues.currency = currencyCode;
+	}
+
+	return productToCustomPost( formValues );
+};
+
+// Thunk action creator to create a new button
+const createPaymentButton = siteId => ( dispatch, getState ) => {
+	const post = getPostFromForm( getState() );
```

Comment:
I would call it `const productCustomPost` so it's clear what we have in mind here.